### DISCUSSION TITLE
How do I use the vpp-agent to change  linux namespace?

### DISCUSSION THREAD

**Discussion by Chise1** [ligato/vpp-agent#1885] [Q&A]
I think I need a more detailed tutorial.
Do I need to connect to vpp to change linux? Or can VP-agent directly connect to linux device and modify namespace?

**Comment by chrismetz09** [comment]
Hi-
Couple of my inputs but seek clarifications from the Ligato dev team. 

1. No, you do not need to connect to VPP to change linux when using the VPP agent.
2. VPP agent uses the linux plugin to handle linux interfaces. I did not test this feature due to machine limitations. Take a look at:

[linux plugins](https://docs.ligato.io/en/latest/plugins/linux-plugins/#linux-interface-plugin). In the configuration examples, you can use `agentctl kvdb put` to configure linux interfaces and their respective namespaces. Would you be able to test this feature and let us know if it works? 

[namespace plugin](https://docs.ligato.io/en/latest/plugins/linux-plugins/#namespace-plugin) - Have not worked with this plugin. I believe it enables namespace management for linux interface plugins.

-chris

**Comment by chrismetz09** [comment]
Also take a look at the linux examples at [Examples](https://docs.ligato.io/en/latest/user-guide/examples/)

**Comment by Chise1** [comment]
Thank you all.
